docs/design.md

```diff
@@ -0,0 +1,104 @@
+ 3. unencrypted with signature, configurable and should be enabled
+ 4. unencrypted without signature, configurable and may be disabled
+
+For confidential containers based on enhanced hardware virtualization technologies (TDX/SEV/PEF etc), there's no easy
+way to share container images among pods/containers running on the same physical node. But it's time consuming to
+download, verify, decrypt and extract container images for each pod/container. So the `image-rs` crate will explore
+technologies and innovations to reduce image preparation time. Data decryption and decompression acceleration libraries
+may be used to speed up image decryption and decompression. Image data on demand loading may also be adopted to reduce
+container startup time. It also helps to share image data among containers/pods with confidentiality and integrity,
+though we still need to figure out the suitable technologies for image data sharing.
+
+### Goals
+ * Support pulling OCI/docker v2 images from registries with or without authentication
+ * Support downloading OCI images from local [OCI layout compatiable](https://github.com/opencontainers/image-spec/blob/main/image-layout.md) storage
+ * Support OCI image decryption
+ * Support OCI image signature verification
+ * Support configurable security policies for allowed OCI images
+ * Support seamlessly integration with Kata Agent and Attestation Agent
+ * Provide a command-line utility for debug
+ * Explore image decryption and decompression acceleration technologies
+ * Explore image on demand loading technologies
+ * Explore technologies to share image data among containers/pods with confidentiality and integrity.
+
+
+### Non-Goals
+ * Image push, content discovery/management operations for remote registries
+ * Image encryption and signing
+ * Image export to local storage
+ * Image import from other service daemon or other local storage format
+
+
+### Existing Solutions
+ * [containers-image-proxy-rs](https://github.com/containers/containers-image-proxy-rs): a rust bindings for skopeo,
+   which doesn't meet our requirements of small TCB and lightweight.
+ * [oci-distribution](https://github.com/krustlet/oci-distribution): a crate from
+   [krustlet](https://github.com/krustlet/krustlet) to run wasm application with K8S. Since wasm is platform agnostic
+   and small code size, `oci-distribution` does not support manifest lists for different platforms and internal data
+   struct is not friend to generic container workload.
+ * [oci-registry-client](https://github.com/ecarrara/oci-registry-client): a very simple client for OCI image registries
+   with limited features and latest commit is one and half years ago.
+
+Above projects focus on image distribution and do not support image unpack and storage management features.
+
+
+## Architecture
+The following diagram demonstrates the overall architecture and how to
+integrate with Kata Agent:
+![Architecture](images/architecture.png)
```

Comment:
As [Kata requires](https://github.com/kata-containers/kata-containers/blob/main/docs/Documentation-Requirements.md#images), I think it makes sense here to include the source for this image to allow future updates without having to either hack the png with a bitmap editor (yuck), or recreate entirely (also yuck ;)

Reply:
Yes, currently the image is from the screenshot of ppt, should we also upload the ppt file?

Reply:
Yes please. It's not an ideal format but since there are open source editors available (openoffice, libreoffice, etc), it should work.

Reply:
attached now :-)